components/script/lib.rs

```diff
@@ -96,3 +96,9 @@ mod window_named_properties;
 
 pub use init::init;
 pub use script_runtime::JSEngineSetup;
+
+pub use crate::dom::bindings::inheritance::HasParent;
+pub use crate::dom::bindings::reflector::{DomObject, MutDomObject, Reflector};
+pub use crate::dom::bindings::trace::CustomTraceable;
+// pub exports for derive macros
```

Comment:
I'm not sure I understand this comment. I'm happy to try to write something here if you can give me a bit of input on what it means. Why is it labeling this export and not the ones above?

Reply:
Ah, running `mach fmt` reordered lines.

When using various derive macros (ex. `JSTraceable`) they impl traits for structs, but to find trait `JSTraceable` they used to use full path. For #1799 we need to refer to these traits from various crates, so full path is different in each crate, and to make it easier for derives will simply expect that appropriate traits will be exported in crate root.

Reply:
Thanks for the background:

```
// These derive macro exports are public, because they are used in the DOM bindings
// and it is useful that they are accessible at the root of the crate.
```

Reply:
These are not derive macros, these are traits (that have same name as derive macros, similar is in rust with `Debug` where you can manually implement `Debug` trait or use `derive(Debug)` (`Debug` derive macro) that will automatically implement `Debug` trait for you) that are used by derive macros (derive macros expand to traits impl) and that's way they need to be exported.

Reply:
I rewrote comment to: `export traits to be available for derive macros`, is that clear enough?

Reply:
How about this:

```

// These trait exports are public, because they are used in the DOM bindings. 
// Since they are used to create derive macros, it is useful that they are accessible at the root of the crate. 
.

```